src/njs/src/njsConnection.cpp

```diff
@@ -798,6 +874,51 @@ void Connection::GetOutBindParams (unsigned short dataType, Bind* bind,
       executeBaton->error= NJSMessages::getErrorMsg(errInvalidBindDataType,2);
       break;
   }
+
+  executeBaton->binds.push_back(bind);
+}
+
+/*****************************************************************************/
+/*
+   DESCRIPTION
+     Processing out binds for PL/SQL indexed table value
+
+   PARAMETERS:
+     dataType - datatype of the bind
+     bind struct, eBaton struct
+*/
+void Connection::GetOutBindParamsArray(unsigned short dataType, Bind *bind, eBaton *executeBaton)
+{
+  Nan::HandleScope scope;
+
+  // Mark bind as an array bind
+  bind->isArray = true;
+
+  // Check for supported bind types
+  switch(dataType)
+  {
+    case DATA_STR :
+      bind->type     = dpi::DpiVarChar;
+      break;
+    case DATA_NUM :
+      bind->type     = dpi::DpiDouble;
+      bind->maxSize  = sizeof(double);
+      break;
+    /*
+    case DATA_DATE :
+      bind->type     = dpi::DpiTimestampLTZ;
+      bind->maxSize  = 0;
+      break;
+    */
+    default :
+      executeBaton->error = NJSMessages::getErrorMsg(errInvalidBindArray, bind->key, "the \"type\" property must be specified and only STRING and NUMBER are (currently) supported");
+      break;
+  }
+
+  // Allocate for OUT Binds of PL/SQL indexed tables variables
+  size_t arrayElementSize = bind->maxSize;
+  Connection::AllocateBindArray(dataType, bind, executeBaton, &arrayElementSize);
```

Comment:
Please move this allocation to Connection::cbDynBufferAllocate ().  For IN  (IN & INOUT) binds from the value, we allocate for C-type and fill in the value(s).  For OUT bind this is deferred to Connection::PrepareAndBind().  Only after OCIStmtPrepare2() we can determine whether the given SQL statement contains RETURNING INTO clause (DML returning).  In this case, the OUT bind will return an array of values.  

It is difficult to consolidate all the allocations to one place, but we can always try in another transaction.

As your drop is allocating for array, moving this allocation to Connection::cbDynBufferAllocate would be appropriate.